supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/statuspolling/JesPollingActor.scala

```diff
@@ -0,0 +1,87 @@
+package cromwell.backend.impl.jes.statuspolling
+
+import akka.actor.{Actor, ActorLogging, ActorRef, PoisonPill, Props}
+import cats.data.NonEmptyList
+import com.google.api.client.googleapis.batch.BatchRequest
+import com.google.api.client.googleapis.batch.json.JsonBatchCallback
+import com.google.api.client.googleapis.json.GoogleJsonError
+import com.google.api.client.http.HttpHeaders
+import com.google.api.services.genomics.model.Operation
+import cromwell.backend.impl.jes.statuspolling.JesApiQueryManager.{JesPollingWorkBatch, JesStatusPollQuery, NoWorkToDo}
+import cromwell.backend.impl.jes.statuspolling.JesPollingActor._
+
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.{Failure, Success, Try}
+import scala.concurrent.duration._
+
+/**
+  * Polls JES for status. Pipes the results back (so expect either a RunStatus or a akka.actor.Status.Failure).
+  */
+class JesPollingActor(pollingManager: ActorRef) extends Actor with ActorLogging {
+
+  implicit val ec: ExecutionContext = context.dispatcher
+
+  override def receive = {
+
+    case JesPollingWorkBatch(workBatch) =>
+      log.debug(s"Got a polling batch with ${workBatch.tail.size + 1} requests.")
+      val batchResultFutures = handleBatch(workBatch)
+      val overallFuture = Future.sequence(batchResultFutures.map(_._2).toList)
```

Comment:
What's this one for?

(I just refactored this so that I could get the List of failures by lifting `overallFuture` up into a `Future[List[Try[Unit]]]`. Would this do the same "for free")?